test/dlblob_test.js

```diff
@@ -0,0 +1,191 @@
+                return callback(null);
+            });
+        var dlblob = new AlAzureDlBlob(mock.DEFAULT_FUNCTION_CONTEXT, testProcessingStub);
+        dlblob.processDlBlobs({}, function(err) {
+            sinon.assert.callCount(deleteBlobStub, 6);
+            sinon.assert.callCount(testProcessingStub, 6);
+            done();
+        });
+    });
+    
+    it('Blob list error', function(done) {
+        // Mock Azure HTTP calls
+        // List blobs
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get('/alertlogic-dl')
+        .query({'restype':'container','comp':'list','maxresults':process.env.DL_BLOB_PAGE_SIZE,'prefix':process.env.WEBSITE_SITE_NAME})
+        .reply(404, mock.CONTAINER_NOT_FOUND);
+        
+        var testProcessingStub = sinon.stub().callsFake(
+                function(context, dlblob, dlblobPayload, callback) {
+                    return callback(null);
+                });
+        var dlblob = new AlAzureDlBlob(mock.DEFAULT_FUNCTION_CONTEXT, testProcessingStub);
+        dlblob.processDlBlobs({}, function(err) {
+            assert.equal(err.code, 'ContainerNotFound');
+            assert.equal(err.statusCode, 404);
+            sinon.assert.callCount(testProcessingStub, 0);
+            done();
+        });
+    });
+    
+    it('Get blob content error', function(done) {
+        // Mock Azure HTTP calls
+        // List blobs
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get('/alertlogic-dl')
+        .query({'restype':'container','comp':'list','maxresults':process.env.DL_BLOB_PAGE_SIZE,'prefix':process.env.WEBSITE_SITE_NAME})
+        .reply(200, mock.LIST_CONTAINER_BLOBS());
+        
+        // Get blob content 2019-01-23T15-53-06Z
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get(/alertlogic-dl\/kktestdl\/ehubgeneral\/2019-01-23T15-53-06Z/)
+        .times(1)
+        .reply(200, mock.GET_BLOB_CONTENT_TEXT);
+        
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .get(/alertlogic-dl\/kktestdl\/ehubgeneral.*/)
+        .times(5)
+        .reply(404, mock.CONTAINER_NOT_FOUND);
+        
+        // Delete blob
+        nock('https://kktestdl.blob.core.windows.net:443', {'encodedQueryParams':true})
+        .delete(/alertlogic-dl.*/)
+        .times(5)
+        .reply(202);
+        
+        var testProcessingStub = sinon.stub().callsFake(
+                function(context, dlblob, dlblobPayload, callback) {
+                    return callback(null);
+                });
+        var dlblob = new AlAzureDlBlob(mock.DEFAULT_FUNCTION_CONTEXT, testProcessingStub);
+        dlblob.processDlBlobs({}, function(err) {
+            assert.equal(err.code, 'ContainerNotFound');
+            assert.equal(err.statusCode, 404);
+            sinon.assert.callCount(testProcessingStub, 1);
```

Comment:
How many ties will deleteBlob be called here, 6? Perhaps worth adding.

Reply:
Good catch. It should be called ones. And it is supposed to be checked here https://github.com/alertlogic/al-azure-collector-js/pull/13/files#diff-ef57d0ffdfcda235d1b919f3630bd9fcR142

I'll change it to 1 time.